### PR TITLE
pkg/obfuscate: Expose more SQL tokeniser state

### PR DESCRIPTION
**Pull request by Taiki-San**
### What does this PR do?

This PR exports some state from the SQL tokeniser enabling the repurposing of the tokeniser outside of simple obfuscation.

### Motivation

The new primitives enable a piece of code to work off tokens, then to export the start/end indices of a token sequence in the original query.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/obfuscate/sql_tokenizer.go [hunk 1]
```diff
@@ -382,7 +382,7 @@ func (tkn *SQLTokenizer) Scan() (TokenKind, []byte) {
 	}
 }
 
-func (tkn *SQLTokenizer) skipBlank() {
+func (tkn *SQLTokenizer) SkipBlank() {
```
Comment (gbbr) [inline pkg/obfuscate/sql_tokenizer.go:385]: Add some documentation please.

**Review by gbbr — COMMENTED** [review]
Can you please provide an example of how you would use the tokenizer with these new methods?

**Comment by Taiki-San** [comment]
Here is a sample code using the tokeniser to parse a query and export tokens:
```
func parseSQL(query string) ([]Token, error) {
	tokenizer := NewSQLTokenizer(query, false, nil)
	tokens := make([]Token, 0)
	for {
		startPos := tokenizer.Offset()
		kind, buff := tokenizer.Scan()
		if kind == EndChar {
			break
		}
		if kind == LexError {
			return nil, fmt.Errorf("couldn't parse `%s` after position %d", query, startPos)
		}

		tokens = append(tokens, Token{
			kind:           kind,
			representation: string(buff),
			start:          startPos,
			end:            tokenizer.Offset(),
		})
		tokenizer.SkipBlank()
	}
	return tokens, nil
}
```
With those `Token`, the higher level logic is able to highlight a section of the query without reformatting the original version after identifying tokens of interest.

**Review by gbbr — APPROVED** [review]
Thank you for providing the example usage. That is very helpful to me in understanding the reasoning behind this. Much appreciated.

**Pull request merged**